src/backend/storage/lmgr/lock.c

```diff
@@ -2053,13 +2072,8 @@ LockSetHoldTillEndXact(const LOCKTAG *locktag)
 		if (!locallock || locallock->nLocks <= 0)
 			continue;
 
-		/*
-		 * GPDB_92_MERGE_FIXME:
-		 * If the lock is taken via the fast-path, locallock->lock would
-		 * be NULL.
-		 * This code should be revised to make GDD work under fast path.
-		 */
-//		locallock->lock->holdTillEndXact = true;
+		if (locallock->lock)
```

Comment:
An assertion is better here.

Reply:
@guofengrichard 
try_heap_open(, NoLock, );
...
relation_close(r, Nolock); --> will leave locallock->lock is NULL in this branch.

However, in master branch of gpdb(not merge branch), locallock is NULL so we never execute the assignment.

I think we could keep it like this.

And refactor it in another MPP story.

@pivotal-ning-yu